test/common.js

```diff
@@ -381,3 +381,7 @@ exports.nodeProcessAborted = function nodeProcessAborted(exitCode, signal) {
     return expectedExitCodes.indexOf(exitCode) > -1;
   }
 };
+
+// Skip test instead of fail when memory constrained
+exports.enoughTestMem = os.totalmem() > 0x20000000; /* 512MB */
+
```

Comment:
Style: drop the blank line at EOF.  Also, can you make this a getter or a functoion?  The way it is now, it calls `os.totalmem()` for every test.  It's not super expensive but it's not free either and little bits add up.